main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala

```diff
@@ -216,8 +217,46 @@ object EntryPoint {
   /**
     * Returns `true` iff `declaredEff` is not `Pure` and contains some non-primitive effect.
     */
-  private def isBadEntryPointEffect(declaredEff: Type) = {
-    declaredEff.effects.exists(sym => !Symbol.isPrimitiveEff(sym))
+  private def isBadEntryPointEffect(declaredEff: Type): Boolean = {
```

Comment:
@magnus-madsen This check is currently duplicated. I almost suggest that we just make it public in `Safety` and then `Entrypoint` calls that function. Then its also clear that `Entrypoint` does safety checking "in advance". Making the special case explicit

Reply:
Can we just move the function to Type?

Reply:
Hmm theyre a bit specfic. They require types that are ground, have no assoc, and no JVM types.

Reply:
But it is a pure function just Type -> Bool right?

Reply:
technically `Type -> Result[Option[CES], err]` or something, but yes

Reply:
oh he whole function, not just the evaluater. yeah that one has a safe signature. i guess we can do that

Reply:
I think phases should be independent.

Reply:
me too, but pretending that they aren't seems less productive then making it explicit

Reply:
A proper fix would be to delay the entrypoint function decoration I think. I has also obscured a couple bugs so it might be the correct choice

Reply:
Sure, but in the interim we can move it to Type?

Reply:
> pretending that they aren't [dependent] seems less productive then making it explicit

without thinking about code - the entrypoint conversion should first check safety before making a new entrypoint. Why not make the code reflect that directly? they somebody can't mess up the local code in entrypoint

Reply:
Does entry point deal with all entry points and not just main?

In any case, if you can collect everything in EntryPoint then that is fine 👍